### PR TITLE
feat(AppUpload): support AppUpload entity [EXT-2641]

### PR DESCRIPTION
**Pull request by FLoppix**
## Summary

Introducing the `AppBundle` entity with the following methods:

- create
- get
- delete

## Description

Added the entity to the flat and nested client and exported all relevant types. Also adjusted the `Organziation` API to support the `AppUpload` entity for the nested client.

## Checklist (check all before merging)

- [x] Both unit and integration tests are passing
- [x] There are no breaking changes
- [x] Changes are reflected in the documentation

When adding a new method:

- [x] The new method is exported through the default and plain CMA client
- [x] All new public types are exported from `./lib/export-types.ts`
- [x] Added a unit test for the new method
- [ ] Added an integration test for the new method
- [x] The new method is added to the documentation

### CODE REVIEW

File: lib/adapters/REST/endpoints/app-upload.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  params: GetOrganizationParams,
+  payload: { file: string | ArrayBuffer | Stream }
+) => {
+  const httpUpload = getUploadHttpClient(http)
+
+  const { file } = payload
+
+  if (!file) {
+    return Promise.reject(new Error('Unable to locate a file to upload.'))
+  }
```
Comment (andipaetzold) [inline lib/adapters/REST/endpoints/app-upload.ts:44]: How could this happen? And do we need an extra check for that?

We also don't check whether the organizationId is an empty string or undefined.

File: lib/adapters/REST/endpoints/app-upload.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+export const del: RestEndpoint<'AppUpload', 'delete'> = (
+  http: AxiosInstance,
+  params: GetAppUploadParams
+) => {
+  const httpUpload = getUploadHttpClient(http)
+
+  return raw.del<void>(httpUpload, getAppUploadUrl(params))
+}
+
+export const create: RestEndpoint<'AppUpload', 'create'> = async (
```
Comment (andipaetzold) [inline lib/adapters/REST/endpoints/app-upload.ts:33]:
```suggestion
export const create: RestEndpoint<'AppUpload', 'create'> = (
```

File: lib/adapters/REST/endpoints/app-upload.ts (hunk 1 above)
Comment (FLoppix) [reply lib/adapters/REST/endpoints/app-upload.ts:44]: Probably not. To be honest I stuck to the implementation we already use for the `Upload` entity.  In my opinion we can remove this and add a check for the `organizationId`.

Comment (andipaetzold) [reply lib/adapters/REST/endpoints/app-upload.ts]: Okay let's leave it as it is for now.

**Review by andipaetzold — APPROVED** [review]

**Pull request merged**